internal/deploy/istioctl/istio.go

```diff
@@ -357,13 +358,13 @@ func unTar(source, target string, deleteSource bool) error {
 }
 
 func unZip(source, target string, deleteSource bool) error {
-	archive, err := zip.OpenReader(filepath.Clean(source))
+	source = filepath.Clean(source)
+	zipReader, err := initReader(source)
 	if err != nil {
 		return err
 	}
-	defer archive.Close()
```

Comment:
Why do we not have to close the zipReader?

Reply:
Because we do not open it anymore, thus it cannot be closed

Reply:
Without opening it anymore, we should not have the problem of resource exhaustion anymore. PLus, we do no need to open it, to extract the zip